src/NLog.Web/NLogRequestPostedBodyHttpModule.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Collections;
+using System.IO;
+using System.Text;
+using System.Web;
+using NLog.Common;
+using NLog.Web.LayoutRenderers;
+
+namespace NLog.Web
+{
+    /// <summary>
+    /// This class is to intercept the HTTP pipeline and to allow additional logging of the following
+    ///
+    /// POST request body
+    ///
+    /// The following are saved in the HttpContext.Items collection
+    ///
+    /// __nlog-aspnet-request-posted-body
+    ///
+    /// To use, subclass this class and set your own Configuration
+    /// </summary>
+    public class NLogRequestPostedBodyHttpModule : IHttpModule
+    {
+        /// <summary>
+        /// The name of the HttpModule
+        /// You may override in the subclass with your own name
+        /// </summary>
+        public string ModuleName { get; set; } = "NLog Request Posted Body Module";
+
+        /// <summary>
+        /// The configuration for the HttpModule
+        /// </summary>
+        public NLogRequestPostedBodyMiddlewareConfiguration Configuration { get; set; } =
+            NLogRequestPostedBodyMiddlewareConfiguration.Default;
+
+
+        /// <summary>
+        /// Hook in to the BeginRequest event to capture the request posted body
+        /// </summary>
+        /// <param name="context"></param>
+        public void Init(HttpApplication context)
+        {
+            context.BeginRequest += InterceptRequest;
+        }
+
+        /// <summary>
+        /// This will forward the necessary arguments to the capture request body method
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        protected void InterceptRequest(object sender, EventArgs args)
```

Comment:
Could this method become private and renamed to `BeginRequest`. I prefer to keep the visibility of methods to absolute minimum, until someone needs access.

Reply:
Completed as requested